### PR TITLE
Allow custom container images

### PR DESCRIPTION
**Pull request by funkypenguin**
Hey guys,

One of our requirements is to be able to build our entire cluster with "pinned" container images pulled from a local repository. This gives us 100% confidence that the images in use have been validated (by us), and gives us the facility to scan all images for vulnerabilities using the "clair" integration with harbor.io

For example, check this out:
```
root@cn1:~# kubectl get pods --all-namespaces -o jsonpath="{..image}" |tr -s '[[:space:]]' '\n' |sort |uniq -c
     28 registry-internal.elpenguino.net/library/calico-cni:v3.8.0
      2 registry-internal.elpenguino.net/library/calico-kube-controllers:v3.8.0
     14 registry-internal.elpenguino.net/library/calico-node:v3.8.0
     14 registry-internal.elpenguino.net/library/calico-pod2daemon-flexvol:v3.8.0
      8 registry-internal.elpenguino.net/library/ceph:v14.2.2-20190722
      2 registry-internal.elpenguino.net/library/cert-manager-cainjector:v0.8.1
      2 registry-internal.elpenguino.net/library/cert-manager-controller:v0.8.1
      2 registry-internal.elpenguino.net/library/cert-manager-webhook:v0.8.1
      6 registry-internal.elpenguino.net/library/coredns:1.3.1
      2 registry-internal.elpenguino.net/library/external-dns:latest
      8 registry-internal.elpenguino.net/library/fluent-bit:1.2.1
     32 registry-internal.elpenguino.net/library/hyperkube:v1.14.2
      2 registry-internal.elpenguino.net/library/kubelet-rubber-stamp-amd64:latest
      2 registry-internal.elpenguino.net/library/metallb-controller:master
      8 registry-internal.elpenguino.net/library/metallb-speaker:master
      4 registry-internal.elpenguino.net/library/nginx-ingress-controller:0.24.1
      2 registry-internal.elpenguino.net/library/rook-ceph:master
      6 registry-internal.elpenguino.net/library/rook-ceph:v1.0.4
root@cn1:~#
```

This PR introduces optional params and logic which allow the operator to explicitly override the container images for all of the core KRIB elements, so that these can be pulled from a local registry. (We maintain an ansible playbook responsible for pulling down images based on sha, retagging, and pushing them into a local harbor.io registry in preparation)

Cheers!
D

### CODE REVIEW

**Comment by galthaus** [comment]
@funkypenguin - you will need to rebase on the latest tree.  I've been addressing some issues for the simple just run me clusters.  For example, kris-dev-reset wouldn't run unless you had consul, vault, and others configured.  The krib-config wouldn't work because krib/master-vip-address or whatever isn't really a determiner of HA - master-count is.  

Please rebase these on top of the latest master.

File: krib/tasks/krib-dev-hard-reset.yaml [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+  Name: vault CA clear
+- Contents: |-
+    #!/bin/bash
+    # Reset vault params
+    set -e
+
+    declare -a WIPE_PARAMS=('vault/unseal-key' 'vault/root-token' 'vault/kms-plugin-token' 'vault/servers' 'vault/servers-done' 'vault/server-ca-pw' 'vault/server-ca-name')
+    declare CLUSTER_TYPE="vault"
+    {{template "setup.tmpl" .}}
+    {{if .ParamExists "krib/cluster-profile" -}}
```
Comment (galthaus) [inline krib/tasks/krib-dev-hard-reset.yaml]: shouldn't this be vault-profile

Comment (funkypenguin) [reply krib/tasks/krib-dev-hard-reset.yaml]: Yes, you're right, it should. It works  by accident, but it should be vault :)

Comment (funkypenguin) [reply krib/tasks/krib-dev-hard-reset.yaml]: Actually, waitasec, no it's not wrong :) krib/cluster-profile is correct - this is not about consul/etc/vault, this is about the name of our krib cluster. It's used throughout KRIB as a method to grab PROFILE_TOKEN

Comment (galthaus) [reply krib/tasks/krib-dev-hard-reset.yaml]: I was just looking at it as you are clearing vault parameters.  The vault parameters live in the vault-profile.  Now that may be the same for profile for all the pieces, but it seems like it should be different.  Unless these are the k8s parameters needed for k8s to use vault.  If so, we should have probably named the vault parameters as krib/vault...  I think these are intended to be the vault parameters though.

Comment (funkypenguin) [reply krib/tasks/krib-dev-hard-reset.yaml]: Yes, they're intended to be vault parameters. FWIY, I don't really understand how the `GenerateProfileToken` magic works, I just know how to copy/paste :) Happy to take your direction on this!

**Pull request merged**